File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/utils/LogicThreadFactory.java
package com.alta_v2.game.gamelogic.utils;

import lombok.RequiredArgsConstructor;

import java.util.concurrent.ThreadFactory;
import java.util.concurrent.atomic.AtomicInteger;

@RequiredArgsConstructor
public class LogicThreadFactory implements ThreadFactory {

    private final AtomicInteger threadCounter = new AtomicInteger();
    private final String threadName;

    @Override
    public Thread newThread(Runnable runnable) {
        Thread thread = new Thread(runnable, this.threadName + this.threadCounter.getAndIncrement());
        thread.setDaemon(true);
        return thread;
    }
}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/utils/LogicThreadFactory.kt
package com.alta_v2.game.gamelogic.utils

import java.util.concurrent.ThreadFactory
import java.util.concurrent.atomic.AtomicInteger

class LogicThreadFactory(private val threadName: String) : ThreadFactory {

    private val threadCounter = AtomicInteger()

    override fun newThread(runnable: Runnable): Thread {
        val thread = Thread(runnable, threadName + threadCounter.getAndIncrement())
        thread.isDaemon = true
        return thread
    }
}